content/hardware/05.pro-solutions/solutions-and-kits/edge-control/tutorials/smart-irrigation-system/content.md

```diff
@@ -0,0 +1,498 @@
+
+`water_volume = PI*(radius)*(radius)*(water_level/100)*1000`
+
+Using flow control variables we store the initial water level as a reference point, any level measured below that point will be considered as water use updating the `water_usage` variable, if we refill the tank, it will save the higher value of water level as a new reference point.
+
+```arduino
+  water_level = vals->water_level_local;
+  water_volume = PI * (radius) * (radius) * ((int)water_level/100) * 1000;
+
+  if (waterCtrl == 0) {
+    currentWater = water_volume;
+    Serial.println("Initial Water Volumen: ");
+    Serial.println(currentWater);
+    water_usage = 0;
+    waterCtrl = 1;
+  }
+
+  if (water_volume < currentWater) {
+    water_usage += currentWater - water_volume;
+    currentWater = water_volume;
+  } else if (water_volume > currentWater) {
+    currentWater = water_volume;
+  }
+```
+
+### The Arduino IoT Cloud Dashboard
+
+Taking advantage of the Arduino IoT Cloud, we can seamlessly integrate a simple but powerful dashboard to monitor and visualize the status of the system in real-time, resulting in a professional Human-Computer Interaction (HCI) as can be seen below:
+
+![Arduino Cloud project dashboard ](assets/dashboard_16-9.png)
+
+Within the Arduino IoT Cloud's dashboard, the system variables can be monitored and controlled. We have temperature, humidity and rain probability gauges showing the current weather status and forecast, toggle switches to control each motorized valve accompanied by a scheduler widget to set automatic irrigation routines, also we have time series graphs showing the daily activated time of each valve. Finally, there are water supply monitoring widgets, one that shows the current water level in percentage from 0-100%, a water volume widget that shows the remaining liters of water in the tank, and a water usage widget that shows the liters used daily.
+
+![Arduino Cloud project dashboard on a smartphone](assets/dashboard-cel.png)
+
+We can easily access this dashboard from a PC, mobile phone or tablet from anywhere, receiving an instantaneous update wherever we are.
+
+## Full Smart Irrigation System Example
+
+All the necessary files to replicate this application note can be found below:
+
+- The complete code can be downloaded [here](assets/Edge-Control_MKR_Codes.zip)
+
+### Project Working Images
```

Comment:
it's great to add the additional media let's try to align better

```suggestion
### The irrigation system working
```

Reply:
Below you can find some additional images and animations showing how the system works: